internal/controller/pkg/revision/runtime_override_options.go

```diff
@@ -396,7 +414,16 @@ func ServiceWithSelectors(selectors map[string]string) ServiceOverride {
 // ServiceWithAdditionalPorts adds additional ports to a Service.
 func ServiceWithAdditionalPorts(ports []corev1.ServicePort) ServiceOverride {
 	return func(s *corev1.Service) {
-		s.Spec.Ports = append(s.Spec.Ports, ports...)
+		names := make(map[string]bool)
+		for _, p := range s.Spec.Ports {
```

Comment:
Same here, for the runtime providers/functions,  the DeploymentRuntimeConfig.spec.serviceTemplate.spec would need to keep the Service port same as 9443 ? with only customizing the targetPort/pod's port. 

Service.spec.
```
ports:
  - port: 9443     ## this can remain the same regardless of hostNetwork used for Providers.
    protocol: TCP
    targetPort: ${customHosNetworktPort}
    name: webhook
```
Changing the Service port (which IMO unnecessary) would require changes in other places like https://github.com/crossplane/crossplane/blob/7ca12206ebf22368933b8e66fc9634f69f0bf1ac/internal/controller/pkg/revision/establisher.go#L364 in the CRDs conversion webhook spec, also..

Reply:
As discussed above, kept this part but added an override to ensure the port will always be the default port (9443).